test/runtests.jl

```diff
@@ -219,4 +219,18 @@ end
             @test M_mul ≈ M * M
         end
     end
+
+    @testset "`pointer`" begin
+        for elem_type ∈ (Int8, Int16, Int32, Int64)
+            for dim_count ∈ 0:4
+                siz = ntuple(Returns(2), dim_count)
+                arr = FixedSizeArray{elem_type, dim_count}(undef, siz)
+                @test pointer(arr) === pointer(arr, 1) != pointer(arr, 2)
```

Comment:
If you want to test some pointer arithmetic you could check that the difference between pointer(arr, 1) and pointer(arr, 2) is the size of elem_type